src/components/calcite-picker-item/calcite-picker-item.tsx

```diff
@@ -0,0 +1,180 @@
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() isSelected = this.selected;
+
+  dir: "rtl" | "ltr";
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.dir = getElementDir(this.el);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerItemSelectedChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggleSelected(coerce?: boolean, emit = false) {
+    this.isSelected = typeof coerce === "boolean" ? coerce : !this.isSelected;
+    if (emit) {
+      this.emitChangeEvent();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  pickerClickHandler = (event: MouseEvent): void => {
+    this.isSelected = !this.isSelected;
+    this.emitChangeEvent(event.shiftKey);
+  };
+
+  secondaryActionContainerClickHandler(event: MouseEvent) {
+    event.stopPropagation();
+  }
+
+  emitChangeEvent(shiftPressed = false) {
+    this.calcitePickerItemSelectedChange.emit({
+      item: this.el,
+      value: this.value,
+      selected: this.isSelected,
+      shiftPressed
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  renderIcon() {
+    const { icon } = this;
+    if (!icon) {
+      return null;
+    }
+    if (icon === ICON_TYPES.grip) {
+      return (
+        <span class={CSS.handle}>
+          <CalciteIcon size="24" path={handleVertical24} />
+        </span>
+      );
+    } else {
+      const path =
+        icon === ICON_TYPES.square
+          ? this.isSelected
+            ? checkSquare16
+            : square16
+          : this.isSelected
+          ? circleFilled16F
+          : circle16F;
+      return (
+        <span class="icon">
+          <CalciteIcon size="16" path={path} />
+        </span>
+      );
+    }
+  }
+
+  render() {
+    const { icon } = this;
+    return (
+      <Host
+        dir={this.dir}
+        class={icon !== ICON_TYPES.square && icon !== ICON_TYPES.circle ? CSS.highlight : null}
+        onClick={this.pickerClickHandler}
+        selected={this.isSelected}
```

Comment:
Why is this set on here? Shouldn't the `selected` prop already handle it?

Reply:
The prop is no longer mutable, so we need to have this here if we want to reflect the state back to the dom. If we DON'T want to reflect the state back to the dom, this can be removed.

Reply:
ok. makes sense.